From e1317f08b9685e6f1cdb290e9d09bf60c94926f6 Mon Sep 17 00:00:00 2001
From: eric zhang <eric.zhang@windriver.com>
Date: Fri, 5 Jan 2018 21:05:50 -0500
Subject: [PATCH] upversion with release num

Signed-off-by: eric zhang <eric.zhang@windriver.com>
---
 SPECS/libibverbs.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/libibverbs.spec b/SPECS/libibverbs.spec
index ce16447..d70ec98 100644
--- a/SPECS/libibverbs.spec
+++ b/SPECS/libibverbs.spec
@@ -12,7 +12,7 @@
 
 Name: libibverbs
 Version: 41mlnx1
-Release: OFED.4.4.2.0.1.44207
+Release: OFED.4.4.2.0.1.44207%{?_tis_dist}.%{tis_patch_ver}
 Summary: A library for direct userspace use of RDMA (InfiniBand/iWARP) hardware
 
 Group: System Environment/Libraries
-- 
2.7.4

