From 09375f6abc530c339c0cd3f3c81d5c5af42abaca Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Fri, 18 Nov 2016 13:53:47 -0500
Subject: libibverbs: fix build failures due to unwanted sgid bit

---
 SPECS/libibverbs.spec | 1 +
 1 file changed, 1 insertion(+)

diff --git a/SPECS/libibverbs.spec b/SPECS/libibverbs.spec
index d70ec98..d3523f9 100644
--- a/SPECS/libibverbs.spec
+++ b/SPECS/libibverbs.spec
@@ -107,6 +107,7 @@ rm -rf $RPM_BUILD_ROOT
 %if %{_with_valgrind}
 %{_libdir}/mlnx_ofed/valgrind/lib*.so
 %endif
+%dir %attr(0755,root,root) %{_includedir}/infiniband
 %{_includedir}/*
 %{_mandir}/man3/*
 %{_mandir}/man7/*
-- 
2.7.4

