From 4a40e5127301444dfcef90e5232585dfcdd8accf Mon Sep 17 00:00:00 2001
From: Don Penney <don.penney@windriver.com>
Date: Mon, 13 Nov 2017 17:22:49 -0500
Subject: [PATCH] Add TIS patches

---
 SPECS/anaconda.spec | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/SPECS/anaconda.spec b/SPECS/anaconda.spec
index 8be37eb..0933f5a 100644
--- a/SPECS/anaconda.spec
+++ b/SPECS/anaconda.spec
@@ -24,6 +24,9 @@ Patch7:	anaconda-centos-skip-retry-if-not-connected.patch
 Patch8: 9800-rpmostreepayload-Rework-remote-add-handling.patch
 Patch9: yumpayload-dont-verify-disabled-repos.patch 
 
+# WRS
+Patch10001: 0001-TIS-Progress-and-error-handling.patch
+
 # Versions of required components (done so we make sure the buildrequires
 # match the requires versions of things).
 %define dbusver 1.2.3
@@ -246,6 +249,9 @@ runtime on NFS/HTTP/FTP servers or local disks.
 %patch8 -p1
 %patch9 -p1
 
+# WRS
+%patch10001 -p1
+
 %build
 %configure --disable-static \
            --enable-introspection \
-- 
2.7.4

