From b8b4612f42aad5760d91f12f6da3dd1b2cc018f1 Mon Sep 17 00:00:00 2001
From: zhipengl <zhipengs.liu@intel.com>
Date: Tue, 14 Aug 2018 07:21:19 +0800
Subject: [PATCH 1/1] Update package versioning for TIS format

Signed-off-by: zhipengl <zhipengs.liu@intel.com>
---
 SPECS/libvirt-python.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/libvirt-python.spec b/SPECS/libvirt-python.spec
index c1fca02..dd0b558 100644
--- a/SPECS/libvirt-python.spec
+++ b/SPECS/libvirt-python.spec
@@ -34,7 +34,7 @@
 Summary: The libvirt virtualization API python2 binding
 Name: libvirt-python
 Version: 4.6.0
-Release: 1%{?dist}%{?extra_release}
+Release: 1%{?_tis_dist}.%{tis_patch_ver}
 Source0: http://libvirt.org/sources/python/%{name}-%{version}.tar.gz
 Url: http://libvirt.org
 License: LGPLv2+
--
2.7.4

