From bb2c1dbc13edd210b5eab8a6221ea718cce8b960 Mon Sep 17 00:00:00 2001
From: Changcheng Liu <changcheng.liu@intel.com>
Date: Wed, 16 Jan 2019 20:31:11 +0800
Subject: [PATCH 01/38] comment code is for ceph-mgr/RESTful plugin

Signed-off-by: Changcheng Liu <changcheng.liu@intel.com>
---
 cephclient/client.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cephclient/client.py b/cephclient/client.py
index 8e9bf17..4de0337 100644
--- a/cephclient/client.py
+++ b/cephclient/client.py
@@ -20,7 +20,7 @@
 #
 
 """
-A ceph-rest-api python interface that handles REST calls and responses.
+A ceph-mgr/RESTful plugin interface that handles REST calls and responses.
 """
 
 import logging
-- 
2.17.1

