From 39196dffa40523da5ef71036337aa2bb76153d31 Mon Sep 17 00:00:00 2001
From: Changcheng Liu <changcheng.liu@intel.com>
Date: Mon, 21 Jan 2019 10:54:10 +0800
Subject: [PATCH 15/38] osd_crush_tree implementation

Signed-off-by: Changcheng Liu <changcheng.liu@intel.com>
---
 cephclient/wrapper.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/cephclient/wrapper.py b/cephclient/wrapper.py
index 49b145c..2bc7243 100644
--- a/cephclient/wrapper.py
+++ b/cephclient/wrapper.py
@@ -342,6 +342,9 @@ class CephWrapper(client.CephClient):
     def osd_crush_rule_ls(self, **kwargs):
         return self.post('request?wait=1', json = {'prefix': 'osd crush rule ls'}, **kwargs)
 
+    def osd_crush_tree(self, **kwargs):
+        return self.post('request?wait=1', json = {'prefix': 'osd crush tree'}, **kwargs)
+
     def osd_dump(self, epoch=None, **kwargs):
         if epoch is not None:
             return self.get('osd/dump?epoch={0}'
-- 
2.17.1

